tests/tor_forward_test.go

```diff
@@ -0,0 +1,53 @@
+package tests
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/cretz/bine/tor"
+	"github.com/cretz/bine/torutil"
+)
+
+func testHandler(w http.ResponseWriter, r *http.Request) {
+	_, err := w.Write([]byte("forward response"))
+	if err != nil {
+		panic(err)
+	}
+}
+
+func TestForwardSimpleHTTP(t *testing.T) {
+	remotePorts := []int{80, 8080}
+
+	// Create a test server listening on a random port
+	server := httptest.NewServer(http.HandlerFunc(testHandler))
+	t.Cleanup(server.Close)
+
+	ctx := GlobalEnabledNetworkContext(t)
+
+	// Forward as an onion service on test ports
+	conf := &tor.ForwardConf{
+		Version3: true,
```

Comment:
Note that v2 services are [past sunset](https://support.torproject.org/onionservices/v2-deprecation/) and current releases of Tor will not allow these to be published. Integration tests which attempt to resolve or publish v2 hidden services are currently failing. I'll have a fix for these in a followup.